src/main/java/com/tangerine/urlshortener/url/controller/dto/ShortenRequest.java

```diff
@@ -0,0 +1,19 @@
+package com.tangerine.urlshortener.url.controller.dto;
+
+import com.tangerine.urlshortener.url.model.vo.Algorithm;
+import com.tangerine.urlshortener.url.model.vo.OriginUrl;
+import com.tangerine.urlshortener.url.service.dto.ShortenParam;
+
+public record ShortenRequest(
+        String originUrl,
+        String algorithm
+) {
+
+    public static ShortenParam to(ShortenRequest request) {
+        return new ShortenParam(
+                new OriginUrl(request.originUrl()),
+                Algorithm.valueOf(request.algorithm())
+        );
+    }
+
+}
```

Comment:
보니까, Service와 Controller의 DTO를 분리하신것 같아요.
Algorithm은 enum이니까 바로 받을 수 있을것 같네요
```suggestion

public record ShortenRequest(
        String originUrl,
        Algorithm algorithm
) {
}
```

또한 Service의 dto를 생성하는데 꼭 정적 팩토리 메소드 일 필요는 없을것 같아요.
자기자신의 static 메소드에 자기자신의 인스턴스를 넘긴다는 어색하네요. 
`무조건 ShortenRequest의 인스턴스`로 ShortenParam을 만들어야 한다면 다음과 같이 사용할 수 있지 않을까요?

```suggestion

public record ShortenRequest(
        String originUrl,
        String algorithm
) {

    public ShortenParam to() {
        return new ShortenParam(
            new OriginUrl(this.originUrl),
            Algorithm.valueOf(this.algorithm)
        );
    }

}
```

그렇다면 컨트롤러에서는?
``` 
    @PostMapping(path = "/shortener")
    public String shortenOriginUrl(
        @RequestBody @Valid  ShortenRequest shortenRequest,
        Model model
    ) {
        UrlMappingResult mappingInfo = urlService.createShortUrl(shortenRequest.to());
        UrlMappingResponse mapping = UrlMappingResponse.of(mappingInfo);
        model.addAttribute("mapping", mapping);
        return "mapping-info";
    }
```

Reply:
DTO 변환을 하나의 형태로 정하고 싶었습니다.
하지만 같은 역할을 한다면 static 보다는 인스턴스 메소드로 구현하는 편이 나은 방법이라고 생각합니다.